api/service/service.go

```diff
@@ -0,0 +1,128 @@
+package service
+
+import (
+	"context"
+	uuid "github.com/satori/go.uuid"
+	"container/list"
+	"fmt"
+	"io"
+	"encoding/json"
+
+	"github.com/spaceuptech/space-api-go/api/proto"
+	"github.com/spaceuptech/space-api-go/api/config"
+	"github.com/spaceuptech/space-api-go/api/model"
+	"github.com/spaceuptech/space-api-go/api/utils"
+)
+
+type CallBackFunction func(string, interface{})()
+type ServiceFunction func(*model.Message, *model.Message, CallBackFunction)()
+
+// Service contains the methods for the service instance
+type Service struct {
+	config  *config.Config
+	service string
+	id      string
+	funcs   map[string]ServiceFunction
+}
+
+func Init(config *config.Config, service string) *Service {
+	id := uuid.NewV1().String()
+	return &Service{config, service, id, make(map[string]ServiceFunction)}
+}
+
+func (s *Service) RegisterFunc(funcName string, function ServiceFunction) {
+	s.funcs[funcName] = function
+}
+
+func (s *Service) Start() {
+	registerRequest := &proto.FunctionsPayload{Service: s.service, Type: utils.TypeServiceRegister, Id: s.id, Project: s.config.Project, Token: s.config.Token}
+	l := list.New()
```

Comment:
Instead of a list use a channel

Reply:
Make the channel within the wrapper for loop, after conn.GetState()

Reply:
Putting it inside the for loop won't work I think.  
The list  just stores the failed requests, to send again (in the next iteration of the loop, when the client reconnects)

Reply:
And since we are not using threads in that part, would it be appropriate to use a channel?

Reply:
Maybe i have got the logic wrong. Lets get on a call and discuss